cpp/test/CMakeLists.txt

```diff
@@ -379,6 +379,22 @@ if(BUILD_TESTS)
     100
   )
 
+  ConfigureTest(
+    NAME
+    NEIGHBORS_ANN_NN_DESCENT_TEST
+    PATH
+    test/neighbors/ann_nn_descent/test_float_uint32_t.cu
+    test/neighbors/ann_nn_descent/test_int8_t_uint32_t.cu
+    test/neighbors/ann_nn_descent/test_uint8_t_uint32_t.cu
+    test/neighbors/ann_nn_descent/test_float_int64_t.cu
```

Comment:
Do we need both uint32_t and int64_t? I don't think the other algorithms instantiate both and this makes the tests take longer to compile. Also didn't you mention we can't use unsigned types for the indexing type since the sign bit needs to be used?

Reply:
I have only kept these types because of the CAGRA accepting any of these types for an input graph. In terms of implementation details, the graph is always built with `int` and then copied back to `uint32_t` or `int64_t`. So I'll just test `int` directly and CAGRA can test the other types indirectly